webapp/tests/karma/unit/services/db-sync.js

```diff
@@ -128,6 +136,23 @@ describe('DBSync service', () => {
       });
     });
 
+    it('error in replication with no docs to send results in "unknown" status', () => {
+      isOnlineOnly.returns(false);
+      Auth.returns(Q.resolve());
+
+      replicationResult = () => Q.reject('error');
+      const onUpdate = sinon.stub();
+      service.addUpdateListener(onUpdate);
+      info.returns(Q.resolve({ update_seq: 100 }));
```

Comment:
Should we add a test which confirms when no replication is required due to update_seq matching?

Reply:
I think we have four states: unknown, required, inprogress and an up-to-date state where no replication is required. I think we have tests for the first three but not the last state, just a simple test for the no-replication-required code path.

Reply:
The update_seq matching is actually the "unknown" state. The up-to-date state is when the replication completes and we don't even bother checking the update_seq. You're right, though, we didn't have a test for this. I've added one.